engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -93,550 +104,349 @@
+        this.minX = minX;
+        this.minY = minY;
+        this.minZ = minZ;
 
-    public BlockRegion copy() {
-        return new BlockRegion(this);
+        this.maxX = maxX;
+        this.maxY = maxY;
+        this.maxZ = maxZ;
+        return this;
     }
 
-    // -- min --------------------------------------------------------------------------------------------------------//
-
     /**
-     * The x-coordinate of the minimum corner
+     * Reset both the minimum and maximum corner of this region.
+     *
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if the given coordinates for min and max are {@link #isValid()
+     *         invalid}.
      */
-    public int minX() {
-        return this.minX;
+    public BlockRegion set(Vector3ic min, Vector3ic max) {
+        return this.set(min.x(), min.y(), min.z(), max.x(), max.y(), max.z());
     }
 
     /**
-     * the minimum coordinate of the first block x
+     * Reset this region to have the same minimum and maximum corner as the {@code source} region.
      *
-     * @return the minimum coordinate x
-     * @deprecated use {@link #minX()}
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if the given coordinates for min and max are {@link #isValid()
+     *         invalid}.
      */
-    @Deprecated
-    public int getMinX() {
-        return this.minX;
+    public BlockRegion set(BlockRegionc source) {
+        return this.set(source.minX(), source.minY(), source.minZ(), source.maxX(), source.maxY(), source.maxZ());
     }
 
-    /**
-     * set the minimum coordinate of the first block x
-     *
-     * @return the minX
-     */
-    public BlockRegion minX(int x) {
-        Preconditions.checkArgument(x <= this.maxX || this.maxX == Integer.MIN_VALUE);
-        this.minX = x;
-        return this;
+    // -- min --------------------------------------------------------------------------------------------------------//
+
+    @Override
+    public int minX() {
+        return this.minX;
     }
 
-    /**
-     * The y-coordinate of the minimum corner
-     */
+    @Override
     public int minY() {
         return this.minY;
     }
 
-    /**
-     * the minimum coordinate of the first block y
-     *
-     * @return the minimum coordinate y
-     * @deprecated use {@link #minY()}
-     */
-    @Deprecated
-    public int getMinY() {
-        return this.minY;
+    @Override
+    public int minZ() {
+        return this.minZ;
     }
 
-    /**
-     * set the minimum coordinate of the first block y
-     *
-     * @return the minY
-     */
-    public BlockRegion minY(int y) {
-        Preconditions.checkArgument(y <= this.maxY || this.maxY == Integer.MIN_VALUE);
-        this.minY = y;
-        return this;
+    @Override
+    public BlockRegion minX(int x, BlockRegion dest) {
+        return dest.set(x, this.minY, this.minZ, this.maxX, this.maxY, this.maxZ);
```

Comment:
A setter like this just looks kind of strange to me. all it does is write to the dest with a substitution of a single value.  I don't really seeing this adding much to the API. 

```
BlockRegion a = new BlockRegion(0,0,0);

BlockRegion b = new BlockRegion(a).minY(-1);
vs
BlockRegion b = a.minY(-1, new BlockRegion(BlockRegion.INVALID));
```